tasks/preflight.yml

```diff
@@ -77,3 +79,39 @@
   register: prometheus_systemd_version
   tags:
     - skip_ansible_lint
+
+- block:
+    - name: Fail if docker labels are set but docker isn't
+      debug:
+        msg: "Labels are set, but the role isn't configured to install prometheus with docker"
+      when:
+        - not prometheus_docker_install
+        - prometheus_docker_labels != []
+
+    - name: Get Docker daemon executable path
+      shell: whereis dockerd | awk '{print $2}'
```

Comment:
No `shell` please. This is an anti-pattern in ansible and should be used only as last resort.